src/Verification.php

```diff
@@ -50,6 +61,25 @@ private function signatureMatches()
         }
     }
 
+    /**
+     * @return bool
+     */
+    private function authorizationMatches()
+    {
+        try {
+            $random = random_bytes(32);
+            return
+              hash_hmac('sha256', $this->expectedAuthorizationBase64(), $random, true) ===
+              hash_hmac('sha256', $this->providedAuthorizationBase64(), $random, true);
```

Comment:
this code formatting is hard to grok. Probably best to assign to variables?

Reply:
Could it be a counter to a [timing attack](https://security.stackexchange.com/questions/83660/simple-string-comparisons-not-secure-against-timing-attacks/83670)?

If so, [hash_equals(string1,string2)](http://php.net/manual/en/function.hash-equals.php) is a timing-attack resistant function, and conveniently we've just bumped minimum PHP version to 5.6 which is the first version this is available:

`return hash_equals($this->expectedAuthorizationBase64(), $this->providedAuthorizationBase64());
`

Will commit and see if it passes the CI for 5.6.

Reply:
Definitely a counter for a timing attack. You learn something new every day :smile: !